### ISSUE TITLE
WARNING no labels found in detect set, can not compute metrics without labels

### ISSUE THREAD

**Issue by MusicYinlinghan** [ultralytics/ultralytics#20595]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

以前都能成功训练，突然就提示没标签

![Image](https://github.com/user-attachments/assets/0b63128d-44e9-4bf6-adc9-ccec18d8f58d)

### Environment

Ultralytics 8.3.131  Python-3.13.3 torch-2.7.0+cu128 CUDA:0 (NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB)
Setup complete  (16 CPUs, 30.8 GB RAM, 248.3/453.9 GB disk)

OS                  Windows-11-10.0.26100-SP0
Environment         Windows
Python              3.13.3
Install             pip
Path                D:\Project\yoloservice\.venv\Lib\site-packages\ultralytics
RAM                 30.79 GB
Disk                248.3/453.9 GB
CPU                 AMD Ryzen 9 8945H w/ Radeon 780M Graphics
CPU count           16
GPU                 NVIDIA GeForce RTX 4060 Laptop GPU, 8188MiB
GPU count           1
CUDA                12.8

numpy                2.1.2>=1.23.0
matplotlib           3.10.3>=3.3.0
opencv-python        4.11.0.86>=4.6.0
pillow               11.0.0>=7.1.2
pyyaml               6.0.2>=5.3.1
requests             2.32.3>=2.23.0
scipy                1.15.3>=1.4.1
torch                2.7.0+cu128>=1.8.0
torch                2.7.0+cu128!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision          0.22.0+cu128>=0.9.0
tqdm                 4.67.1>=4.64.0
psutil               7.0.0
py-cpuinfo           9.0.0
pandas               2.2.3>=1.1.4
seaborn              0.13.2>=0.11.0
ultralytics-thop     2.0.14>=2.0.0

### Minimal Reproducible Example

data.yaml
/*
train: D:\Project\yoloservice\dataset\train\images
val: D:\Project\yoloservice\dataset\val\images

nc: 1
names: ['fire']
*/

CLI
yolo detect train data=D:\Project\yoloservice\data.yaml model=D:\Project\yoloservice\yolo11x.pt epochs=2 imgsz=640 batch=8 device=0 project=D:\Project\yoloservice\my_train name=exp1

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MusicYinlinghan, thank you for reaching out to Ultralytics 🚀! This is an automated response to help guide you while an Ultralytics engineer will also assist you soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for helpful guidance, including many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions about dataset structure and training are answered there.

If this is a 🐛 Bug Report, please ensure you have provided a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) (thank you for including your `data.yaml` and CLI command!). If possible, double-check your label files in your `labels` directories to confirm they match your images and are not empty or missing, as this can cause the "no labels found" warning.

For custom training ❓ questions, please include additional information such as sample images, label files, and any training logs to help us assist you more efficiently. Make sure to follow our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, check out [Discourse](https://community.ultralytics.com/). Or join our [Subreddit](https://reddit.com/r/Ultralytics) to connect and share knowledge.

## Upgrade

Please upgrade to the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to confirm if your issue persists in the current version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. These tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your detailed report and for helping improve Ultralytics! An engineer will follow up with you soon.

**Comment by Y-T-G** [comment]
Are there labels in `val` folder?
